crates/sui-core/src/authority.rs

```diff
@@ -1104,6 +1105,34 @@ impl AuthorityState {
         SuiTransactionEffects::try_from(effects, self.module_cache.as_ref())
     }
 
+    pub async fn dev_inspect_transaction(
+        &self,
+        transaction: TransactionData,
+        transaction_digest: TransactionDigest,
+    ) -> Result<DevInspectResults, anyhow::Error> {
+        let (gas_status, input_objects) =
+            transaction_input_checker::check_transaction_input(&self.database, &transaction)
```

Comment:
From what I can tell this implies that the transaction needs a valid gas object, and this still verifies the object ownership but for dev inspect we want to bypass those checks

Reply:
We do yeah. I can try to fix that in this PR. I basically just want to check that all the objects exist. Would be nice to just strait up remove the gas logic too, but not sure if that is feasible. Any ideas?

Reply:
I wonder if we can make `gas_payment` on the transaction an `Option<ObjectRef>`, and just enforce it to be some for non dev-inspect transactions?

Reply:
A good suggestion, which gives me an idea! We are also looking at making `gas_payment` be a vector of `ObjectRef` (for a different work stream). Once we do that, we can make `gas_payment` optionally empty for the `dev-inspect` case? (though it might be tricky for the execution engine)